### DISCUSSION TITLE
Getting URL image

### DISCUSSION THREAD

**Discussion by mtm-aunglinnnaing** [python-pillow/Pillow#6960] [Q&A] [answered]
Can I get image URL after writing text to image(image url)? Please ..

**Comment by radarhere** [comment, marked as answer]
Hi. I don't understand your question.

1) Are you asking how to reread text after you have written it to an image? In other words, OCR? Pillow doesn't offer that functionality. Perhaps try another Python library, like https://pypi.org/project/pytesseract/
2) Are you asking how to encode your image using base64, so that it can then be part of a `background-image` CSS property?
```python
import base64
with open("im.png", "rb") as fp:
    data = base64.b64encode(fp.read())
    print(b"background-image: url(data:image/gif;base64,"+data+b")")
```
You mentioned that you wanted to write on the image first though.
```python
import base64
from io import BytesIO
from PIL import Image, ImageDraw, ImageFont
im = Image.new("RGB", (290, 65))
font = ImageFont.truetype("FreeMono.ttf", 40)
d = ImageDraw.Draw(im)
d.text((10, 10), "Hello World", font=font)

b = BytesIO()
im.save(b, "PNG")
data = base64.b64encode(b.getvalue())
print(b"background-image: url(data:image/gif;base64,"+data+b")")
```
3) Are you asking how to download an image from a URL?
```python
import urllib.request
urllib.request.urlretrieve("https://python-pillow.org/images/pillow-logo-light-text-1280x640.png", "pillow.png")
```
4) Are you asking how to upload an image to the Internet? This is unfortunately too vague to answer helpfully.

If it would be helpful just to know how to write text onto an image, you can use part of what I wrote above.

```python
from PIL import Image, ImageDraw, ImageFont
im = Image.new("RGB", (290, 65))
font = ImageFont.truetype("FreeMono.ttf", 40)
d = ImageDraw.Draw(im)
d.text((10, 10), "Hello World", font=font)
```

**Reply by radarhere** [reply]
@scm-aunglinnaing is any of the above helpful? If not, could you rephrase your question with some more detail?
